src/opam.nix

```diff
@@ -128,9 +128,13 @@ in rec {
     builder ({ inherit src name version; } // importOpam opamFile) resolveEnv;
 
   listRepo = repo:
-    mergeSortVersions (map (p: listToAttrs [ (nameVerToValuePair p) ])
-      (concatMap attrNames
-        (attrValues (readDirRecursive (repo + "/packages")))));
+    if builtins.pathExists (repo + "/packages")
+      then 
+        mergeSortVersions (map (p: listToAttrs [ (nameVerToValuePair p) ])
+          (concatMap attrNames
+            (attrValues (readDirRecursive (repo + "/packages")))))
+      else
+        {};
```

Comment:
```suggestion
    optionalAttrs (pathExists (repo + "/packages"))
        (mergeSortVersions (map (p: listToAttrs [ (nameVerToValuePair p) ])
          (concatMap attrNames
            (attrValues (readDirRecursive (repo + "/packages"))))));
```

Reply:
looks like that should work. I'll test it to be sure.